src/utils/api/pm/index.tsx

```diff
@@ -34,25 +36,37 @@ const addWallet = async (
           Bearer: "Bearer " + bearer,
           walletRequest
         }),
-      (_e) => {
-        onError(ErrorsType.CONNECTION);
-        return toError;
-      }
+      (_error) => toError
     ),
     TE.fold(
-      (_e) => async () => onError(ErrorsType.SERVER),
+      () => async () => onError(ErrorsType.GENERIC_ERROR), // When the promise rejects
       (resp) => async () =>
         pipe(
           resp,
           E.fold(
-            () => onError(ErrorsType.GENERIC_ERROR),
+            (_errors) => onError(ErrorsType.GENERIC_ERROR), // When errors
             ({ status, value }) => {
-              const idWallet = value?.data?.idWallet;
-              if (status === 200 && idWallet) {
-                onSuccess(idWallet);
-              } else {
-                onError(ErrorsType.GENERIC_ERROR);
-              }
+              pipe(
+                status,
+                utils.validators.evaluateHTTPfamilyStatusCode,
+                O.match(
+                  () => onError(ErrorsType.GENERIC_ERROR),
+                  utils.validators.matchHttpFamilyResponseStatusCode(
+                    {
+                      "2xx": () => {
+                        const idWallet = value?.data?.idWallet;
+                        return idWallet
+                          ? onSuccess(idWallet)
+                          : onError(ErrorsType.GENERIC_ERROR);
+                      },
+                      "4xx": () =>
+                        // eslint-disable-next-line functional/immutable-data
+                        (window.location.href = `${API_HOST}${API_PM_BASEPATH}/v3/webview/logout/bye?outcome=0`)
```

Comment:
great! just a consideration, with 4xx the outcome should be 1 (GENERIC_ERROR) according to https://editor.swagger.io/?url=https://raw.githubusercontent.com/pagopa/pagopa-infra/7b9c99d03f412fcd56c56b0563969c89cd33ba02/src/domains/wallet-app/api/webview-payment-wallet/v1/_openapi.json.tpl

Reply:
thank you!